### PR TITLE
Fixed more RFS version* SonarQube

### PR DESCRIPTION
**Pull request by chelma**
### Description
* Last fixes for the RFS version* SonarQube violations, assuming some updates to the exclusion list which will be forthcoming

Screenshot 2024-09-19 at 4 44 13 PM

### Issues Resolved
* https://opensearch.atlassian.net/browse/MIGRATIONS-2006

### Testing
All the local tests

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RFS/src/main/java/com/rfs/version_os_2_11/GlobalMetadataCreator_OS_2_11.java [hunk 1]
```diff
@@ -131,23 +140,29 @@ private List<String> createTemplates(
...
+        ) {
+
+        List<String> templateList = new ArrayList<>();
+
         templatesToCreate.forEach((templateName, templateBody) -> {
             log.info("Creating {}: {}", templateType, templateName);
 
             if (mode == MigrationMode.SIMULATE) {
-                var alreadyExists = templateType.alreadyExistsCheck.templateAlreadyExists(client, templateName);
-                if (!alreadyExists) {
```
Comment (peternied) [inline RFS/src/main/java/com/rfs/version_os_2_11/GlobalMetadataCreator_OS_2_11.java:150]: Another rule that IMO makes the code harder to read, but oh well.  We can review the rules in depth later

**Review by peternied — APPROVED** [review]
Thanks for cleaning these up.

File: RFS/src/main/java/com/rfs/version_os_2_11/GlobalMetadataCreator_OS_2_11.java (hunk 1 above)
Comment (chelma) [reply RFS/src/main/java/com/rfs/version_os_2_11/GlobalMetadataCreator_OS_2_11.java:150]: Yeah, I'm not crazy about a fair number of them.  I suspect it's a feature, not a bug, that disabling rules is so hard though.

**Review by peternied — APPROVED** [review]
Thanks for unblocking the gradle workflow

**Pull request merged**